### DISCUSSION TITLE
Running with medium container - apt-get fails to install a stock package

### DISCUSSION THREAD

**Discussion by ghenry** [nektos/act#1007] [Q&A] [answered]
Hi all,

I'm trying to run this, which runs fine on GH Actions:

https://github.com/SentryPeer/SentryPeer/blob/main/.github/workflows/clang-analyzer.yml

like so:

`act -j clang-analyzer`

and it fails at the apt-get stage with `autoconf-archive` saying it's not available, but is referred to by another package. I thought this would run exactly like GH Actions? It is a stock package in Ubuntu 20.

Do I need the full 20GB image? What did I miss?

Thanks,
Gavin.

**Comment by catthehacker** [comment, marked as answer]
missing `apt-get update` in workflow

**Comment by ghenry** [comment]
Ah, the large must be up to date a bit then. I do do it here https://github.com/SentryPeer/SentryPeer/blob/main/.github/workflows/clang-analyzer.yml#L35

Good catch! It's nice having act to spot this!

**Reply by catthehacker** [reply]
You need to update indexes before installing packages
https://docs.github.com/en/actions/using-github-hosted-runners/customizing-github-hosted-runners#installing-software-on-ubuntu-runners
![image](https://user-images.githubusercontent.com/31106839/154936595-67f61be4-010f-40b8-a64a-a3ed73a8a50b.png)

**Reply by catthehacker** [reply]
It applies to Ubuntu stock packages as well. It works on GitHub because the image is rebuilt and replaced every week but that's not supported scenario either (and it's something people aren't finding out until there is stale index).

**Reply by ghenry** [reply]
Yep, going to update all my workflows. Glad I started using act! Thanks for your help.

**Reply by hholst80** [reply]
The base image should not even have an index to begin with... 

Now for the the age old question. To apt get upgrade or not to apt get upgrade...? 🤔😆
